diesel_cli/src/main.rs

```diff
@@ -259,6 +263,41 @@ fn connection(database_url: &str) -> PgConnection {
         .expect(&format!("Error connecting to {}", database_url))
 }
 
+// Converts an absolute path to a relative path, with the restriction that the
+// target path must be in the same directory or above the current path. We
+// iterate over both the target and the current path, doing nothing if the
+// path parts are the same. The first time the parts are different, we know we
+// just passed the last common ancestor, so everything left in the current
+// path needs to get converted to "../", and everything in the target path
+// gets appended after the up path.
+fn convert_absolute_path_to_relative(target_path: &PathBuf, current_path: &PathBuf)
+    -> PathBuf
+{
+    let mut target_path_iter = target_path.iter();
```

Comment:
That could be used to generate the "down" part of the path after we've figured out the common ancestors, yeah. I think it would just replace line 294-296